### PR TITLE
Move CommonCrypto and zlib dependencies mapping to custom module map

### PR DESCRIPTION
**Pull request by turbulem**
Details:
https://github.com/daltoniam/Starscream/pull/520

### CODE REVIEW

**Comment by fassko** [comment]
Can you look into why it doesn't build with Swift PM. I get this error:

```
Compression.swift:30:8: error: no such module 'Starscream.SSCZLib'
import Starscream.SSCZLib
```

**Comment by turbulem** [comment]
Sure, let me check

**Comment by turbulem** [comment]
Looks like proposed solution cannot be adopted by SPM. It doesn't seem to support explicit modules. I will investigate what can be done here.

**Comment by nuclearace** [comment]
I would see for SPM if you could just use Swift 4.2's system targets. They look something like this https://github.com/nuclearace/SwiftDiscord/tree/vapor3/Sources/Sodium and https://github.com/nuclearace/SwiftDiscord/blob/vapor3/Package.swift#L43

**Comment by turbulem** [comment]
@nuclearace Thank you, it looks very relevant.
Also I think that in iOS12/Mojave CommonCrypto has system bindings in swift so if target is the latest SDK I can get rid of Obj-C helper

**Comment by fassko** [comment]
Maybe this is the option we can aim. You can still use older release for lover Swift version.

**Comment by fassko** [comment]
@turbulem looks like you're right about CommonCrypto https://stackoverflow.com/a/50690346

**Comment by fassko** [comment]
In which version of Swift are these libraries included? I think need to explicitly set Swift 4.2 version in Package.swift and podspec file. .swift-version is going to be deprecated and isn't used anymore?

File: Starscream.podspec [hunk 1]
```diff
@@ -11,11 +11,10 @@ Pod::Spec.new do |s|
   s.osx.deployment_target = '10.10'
   s.tvos.deployment_target = '9.0'
   s.watchos.deployment_target = '2.0'
-  s.source_files = 'Sources/*.swift'
-  s.libraries    = 'z'
+  s.source_files = 'Sources/**/*.{h,m,swift}'
+  s.module_map = 'Sources/modulemap/Starscream.modulemap'
+  s.private_header_files = 'Sources/modulemap/**/*.h'
```
Comment (fassko) [inline Starscream.podspec]: If we delete modulemap and header file then this is not needed anymore.

Test also with some sample project integrating this pod against https://websocket.org/echo.html

File: Package.swift [hunk 2]
```diff
@@ -27,10 +27,7 @@ let package = Package(
         products: [
```
Comment (fassko) [inline Package.swift]: I think let's set Swift 4.1 here as well likewise in podspec.

**Comment by turbulem** [comment]
@fassko I updated my solution. It only works with new SDK, which might not be the best option. However it solves the problem and seem to work fine with Pods/Carthage/SPM.

**Pull request merged**